src/relativedeltafield/fields.py

```diff
@@ -76,11 +76,15 @@ def get_db_prep_value(self, value, connection, prepared=False):
         if value is None:
             return value
         else:
-            eng = settings.DATABASES[self.model._meta.default_manager.db]['ENGINE']
-            if 'postgres' in eng:
+            model = getattr(self, 'model', None)
+            if model:
+                eng = settings.DATABASES[model._meta.default_manager.db]['ENGINE']
+                if 'postgres' in eng:
+                    return format_relativedelta(self.to_python(value))
+                else:
+                    return relativedelta_as_csv(self.to_python(value))
+            else:  # We pray that this is postgres!
```

Comment:
Can't you use `connection.vendor` like you do elsewhere? I suppose that's the reason this method receives the connection as an argument. Should make the code simpler as well.

Reply:
You're abslutely right! I must have been very tired. WIll commit a fix